index.js

```diff
@@ -141,44 +193,59 @@ const line = []
 
 
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 1", "color: red")
+console.log("%cScope & Closures - Question 1", "color: red")
+
+const otherNewLine = []
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 1`
+function takeATicketNumber(line) {
+otherNewLine.push(1)
+line.push(otherNewLine.length)
+return `Welcome. You are ticket number ${otherNewLine.length}`
+newLine += 1
+}
+
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 1`
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 2`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 2`
 
-// console.log(nowServing(line))
-// // => `Currently serving 1.`
+console.log(nowServing(line))
+// => `Currently serving 1.`
 
-// console.log(nowServing(line))
-// // => `Currently serving 2.`
+console.log(nowServing(line))
+// => `Currently serving 2.`
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 3`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 3`
 
-// console.log("%c----------", "color: red") 
+console.log("%c----------", "color: red") 
 
 // ***** Scope & Closures - Question 2 *****
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 2", "color: red")
+console.log("%cScope & Closures - Question 2", "color: red")
 
-// const newLine = []
-// const takeATicketNumberFunc = ticketNumberGeneratorFunc()
-// console.log(takeATicketNumberFunc(newLine))
-// // => `Welcome. You are ticket number 1`
+// Quite confused by this bonus question tbh 
+// pls explain a bit thank you!!!!!
+
+function ticketNumberGeneratorFunc() {
+  return takeATicketNumber
```

Comment:
You're on the right path in that this function returns a function.
However, the `takeATicketNumber` is still using the global variable of `otherNewLine`.
Take a look at the solution in the solution branch, but the general idea is that JS as a language treats functions as objects, so your solution might look something like this:
```
function ticketNumberGeneratorFunc() {
  // ...
  return function(){
  }
}
```